src/Components/DataHistoryExplorer/DataHistoryExplorer.tsx

```diff
@@ -39,24 +80,371 @@ const DataHistoryExplorer: React.FC<IDataHistoryExplorerProps> = (props) => {
+            });
+        },
+        [dispatch]
+    );
+
+    const handleOnRemoveSeriesClick = useCallback(
+        (seriesId: string) => {
+            const selectedIdx = state.timeSeriesTwins.findIndex(
+                (tsTwin) => tsTwin.seriesId === seriesId
+            );
+            dispatch({
+                type: DataHistoryExplorerActionType.REMOVE_TIME_SERIES_TWINS,
+                payload: { seriesIdx: selectedIdx }
+            });
+            const telemetry =
+                TelemetryEvents.Tools.DataHistoryExplorer.UserAction
+                    .RemoveSeries;
+            sendDataHistoryExplorerUserTelemetry(telemetry.eventName, [
+                {
+                    [telemetry.properties.itemIndex]: selectedIdx
+                }
+            ]);
+            /**
+             * when a series is removed, also remove if from usedColors reference
+             * so that whenever adding a new series we can use these available colors first
+             * before picking the next color in Highcharts palette
+             */
+            usedSeriesColorsRef.current.splice(
+                usedSeriesColorsRef.current.findIndex(
+                    (c) =>
+                        c ===
+                        state.timeSeriesTwins.find(
+                            (tsTwin) => tsTwin.seriesId === seriesId
+                        ).chartProps.color
+                ),
+                1
+            );
+        },
+        [state.timeSeriesTwins, dispatch, sendDataHistoryExplorerUserTelemetry]
+    );
+
+    const handleOnViewModeChange = useCallback(
+        (viewerMode: TimeSeriesViewerMode) => {
+            dispatch({
+                type: DataHistoryExplorerActionType.SET_VIEWER_MODE,
+                payload: { viewerMode: viewerMode }
+            });
+            const telemetry =
+                TelemetryEvents.Tools.DataHistoryExplorer.UserAction.ChangeView;
+            sendDataHistoryExplorerUserTelemetry(telemetry.eventName, [
+                {
+                    [telemetry.properties.view]: viewerMode
+                }
+            ]);
+        },
+        [dispatch, sendDataHistoryExplorerUserTelemetry]
+    );
+
+    const handleOnChartOptionChange = useCallback(
+        (chartOptions: ITimeSeriesCommandBarOptions) => {
+            dispatch({
+                type:
+                    DataHistoryExplorerActionType.SET_COMMAND_BAR_CHART_OPTIONS,
+                payload: { chartOptions: chartOptions }
+            });
+            logDebugConsole(
+                'debug',
+                `Chart options changed: ${JSON.stringify(chartOptions)}`
+            );
+            const telemetry =
+                TelemetryEvents.Tools.DataHistoryExplorer.UserAction
+                    .ChangeChartOption;
+            sendDataHistoryExplorerUserTelemetry(telemetry.eventName, [
+                {
+                    [telemetry.properties.chartOptions]: chartOptions
```

Comment:
great idea to capture these, just make sure we can't end up with anything sensitive or data from their chart in this like a series name or something that might be related to their actual twin.

Reply:
yeah, there should be no sensitive information related to series, only their "count" and if they are set to "casted to numeric". I will add the id for the currently opened modal to tie all the user interactions in a later PR, it is in my list to do as the final wrap up PR.